### ISSUE TITLE
[RPC] Method "torrent-set" ignores "location".

### ISSUE THREAD

**Issue by killemov** [transmission/transmission#4726]
### What is the issue?

Aside from having a different `torrent-get` (`downloadDir`) from `torrent-set` (`location`) argument, the setter argument still seems to be ignored. So an extra `torrent-set-location` call needs to be made. I am looking to scrap that extra call for TM 4.00. Maybe unify `downloadDir` and `location` somehow as well?

From [Shift](https://github.com/killemov/Shift/blob/eb004cc6d969c91cd7a2f4bcd9a6eacddd709cde/shift.js#L2779):
```javascript
if( data.hasOwnProperty( "downloadDir" ) ) {
  data[ "location" ] = data[ "downloadDir" ];
  delete data[ "downloadDir" ];
  doRequest( "torrent-set-location", { ids: [ torrent.id ], location: data[ "location" ], move: moveLed.value } );
}
```

This might be related to #3337 as well.

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

3.00 (bb6b5a062e)

**Comment by ckerr** [comment]
> Aside from having a different torrent-get (downloadDir) from torrent-set (location) argument, the setter argument still seems to be ignored. So an extra torrent-set-location call needs to be made.

I agree that it's unclear in the doc. The problem with removing `torrent-set-location` is that it loses the ability to set the `move` flag. Maybe `torrent-set: location` should be removed from the doc.

> I am looking to scrap that extra call for TM 4.00.

Unfortunately that's not going to happen for 4.0.0. It's scheduled to be released tomorrow, so we're way past the point of any API redesign. This will need to be addressed in another semver major or minor release. More discussion on release timing is [here](https://github.com/transmission/transmission/discussions/3803#discussioncomment-4729033).

By the way, while you're here please respond to https://github.com/transmission/transmission/issues/3226#issuecomment-1412506099

**Comment by ckerr** [comment]
> Aside from having a different torrent-get (downloadDir) from torrent-set (location) argument, the setter argument still seems to be ignored. So an extra torrent-set-location call needs to be made.

I agree that it's unclear in the doc. The problem with removing `torrent-set-location` is that `torrent-set: location` would lose the ability to set the `move` flag. Maybe `torrent-set: location` should be removed from the doc.

> I am looking to scrap that extra call for TM 4.00.

Unfortunately there's not time for any more API redesign for 4.0.0, which is scheduled to be released tomorrow. This will need to be addressed in another semver major or minor release. More discussion on release timing is [here](https://github.com/transmission/transmission/discussions/3803#discussioncomment-4729033).

By the way, while you're here please respond to my repeated requests for you to clarify the "lot of direct documentation [that] got lost when rpc-spec.txt was transformed to rpc-spec.md" at  https://github.com/transmission/transmission/issues/3226#issuecomment-1412506099

**Comment by killemov** [comment]
The problem is not with `torrent-set-location` and `torrent-set` + `location` coexisting but with `torrent-set` + `location` not working. The `move` flag should be inferred false (source content is not there OR destination content already there) or just always true. If for now `downloadDir` and `location` could be aliased internally for `torrent-get` and exposed in the RPC then we could drop the least preferable on which is ... downloadDir?

**Comment by ckerr** [comment]
I wonder if libtransmission could figure out the `move` flag on its own just by looking at the src & dst folders....
